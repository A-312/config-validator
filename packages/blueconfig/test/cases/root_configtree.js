'use strict';

const blueconfig = require('blueconfig');
const LISTOFERRORS = require('blueconfig/lib/error.js').LISTOFERRORS;

function isObjNotNull(obj) {
  return typeof obj === 'object' && obj !== null;
}

exports.formats = {
  children: {
    validate: function(children, schema, fullname) {
      const errors = [];

      if (!isObjNotNull(children)) {
        throw new Error('must be of an Object not null');
      }

      Object.keys(children).forEach((keyname) => {
        try {
          const conf = blueconfig(schema.children).merge(children[keyname]).validate();
          this.set(keyname, conf.getProperties());
        } catch (err) {
          err.parent = fullname + '.' + keyname;
          errors.push(err);
        }
      });

      if (errors.length !== 0) {
        throw new LISTOFERRORS(errors)
      }
    }
  }
};

exports.conf = {
  format: 'children',
  default: {},
  children: {
    // conf.{country-name}.xxx:
    name: {
      format: 'String',
      default: undefined,
      required: true
    },

    population: {
      format: 'int',
      default: 0
    },

    subregion: {
      format: 'String',
      default: 'Europe'
    }
  }
};

exports.data = {
  germany: {
    name: 1,
    population: '83783942 persons',
    subregion: 'Western Europe'
  },
  france: {
    name: 'France',
    population: 65273511
  },
  italy: {
    name: 'Italy',
    population: 60461826,
    subregion: 2
  }
};
