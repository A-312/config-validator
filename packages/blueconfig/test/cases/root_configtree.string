{
  "germany": {
    "name": 1,
    "population": "83783942 persons",
    "subregion": "Western Europe"
  },
  "france": {
    "name": "France",
    "population": 65273511,
    "subregion": "Western Europe"
  },
  "italy": {
    "name": "Italy",
    "population": 60461826,
    "subregion": 2
  }
}