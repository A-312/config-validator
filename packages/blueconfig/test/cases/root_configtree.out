Validate failed because wrong value(s):
  - root: Custom format "children" tried to validate something and failed:
    1) germany:
      - name: must be of type String: value was 1
    2) italy:
      - subregion: must be of type String: value was 2, getter was `value`